pkg/storage/image.go

```diff
@@ -407,7 +408,11 @@ func (svc *imageService) PullImage(systemContext *types.SystemContext, imageName
 		return nil, err
 	}
 	if options == nil {
-		options = &copy.Options{}
+		options = &copy.Options{
+			SourceCtx: &types.SystemContext{
+				AuthFilePath: svc.globalAuthFile,
+			},
```

Comment:
Looking at current non-test callers, this will never trigger AFAICS; both are passing a non-nil `options`.

(And if `svc.globalAuthFile` is effectively never used, it should not exist.)

There are _quite a few_ other callers that refer to the image via `ImageReference.NewImage{,Source}`; shouldn’t all of them also have the `AuthFilePath` value set up, at some point?

I don’t quite know how to think about the conceptual separation between `ImageServer` and its callers; still, the current placement of the `AuthFilePath` option seems incomplete/inconsistent.

Reply:
The main place we need this authFile to be set is when pulling down images right? I was looking at the other callers of `ImageReference.NewImage{}` and don't think those need AuthFilePath value set up. If you are worried about a particular part, can you point me to it in case I missed it.

Reply:
https://github.com/cri-o/cri-o/pull/2510

But, wow, what a maze.  The `alltransports.ParseImageName` in `imageService.getRef` opens up many… possibilities.